content/0008-standalone-workflow-execution/rfc.md

```diff
@@ -0,0 +1,294 @@
+  directory.
+    - We check if a known workflow directory exists
+    - If the directory exists, all YAML files are ran against the cluster
+
+The workflow directory hook is handy so we can further bootstrap a development
+environment for the SaaS or make sure any resources we want will automatically
+exist on the cluster without complicating `relay` any futher. We just let the
+cluster utilize its own ability to run workflows.
+
+#### Image management
+
+The key to `relay` being useful is proper container image management. This is a
+feature that allows a developer to easily write code locally and run it against
+a cluster without committing and waiting for a continuous integration cycle.
+
+Container images for Relay services are initially pulled using the `latest` tag
+and retagged to `local` (e.g. `relay-operator:local`) unless a `local` tag already
+exists for that image.  These local images are then imported into the cluster by
+`relay`.
+
+When a development build script for a Relay service is ran, the resulting
+docker image is tagged using `local` and if the `relay` utility exists in `$PATH`,
+a call to `relay cluster images import` is made to instruct `relay` to import the new
+image. Unrelated to image management, the build script can then trigger a
+workflow run for the Relay service in question.
+
+#### Persisted configuration
+
+In order to start and stop cluster, and manage deployments and users, `relay`
+accesses all the data it needs from a `config.json` file that is populated when
+the cluster is created. This contains the kubeconfig, database credentials,
+vault credentials, helm values, and references to deployment images.
+
+This file is the source of authority for all cluster configuration. Changes made
+to it will reflect down into anything else that exists for the cluster on disk
+(such as the helm values.yaml). This being said, `relay` will not use this file to
+update things like database and vault credentials. Any changes there will need
+to be done manually.
+
+## Engineering-level explanation
+
+While versions beyond V1 have not been planned, I can see requests for changes
+to this system once it's put to use by the team and we should plan for bigger
+changes using extension RFCs to this one. Below I've outlined rough steps that
+should get us to V1.
+
+### V1: Opinionated and just works
+
+The first version should not allow for much configuration. It should provide the
+developer with a working cluster that can update Relay service deployments with
+new container images and run workflows without having to POST them to an
+instance of `relay-api`.
+
+In this version, we will add new commands to the Relay CLI. These commands will
+utilize packages from K3d, Kubernetes and other tools in the space that do a lot
+of the heavy lifting already.
+
+This version will require the developer to install a couple hard dependencies in
+order for `relay cluster` to function:
+
+* kubectl
+* docker
```

Comment:
A few things I think would be worthy of additional commentary here:

1. What's the process to install non-core software that's necessary for the SaaS API/UI to function correctly? For example, how do we get PostgreSQL and Redis installed after the minimal dev environment is running?
2. I think I see a path to stand up system tests with this approach and we kicked around a few ideas in this regard (e.g., putting a bunch of Kubernetes manifests in a directory and having the cluster automatically suck them in), but it might be worth having a small addendum to explain.
3. Does this encompass having a real `Workflow` (not `WorkflowRun`) CRD type? For example, will we expect `relay workflow add` to work with this change, or simply be disabled for the non-`relay.sh` context?